feat(trace-view): Add documentation for the new Trace View (#3347)
getsentry/sentry-docs: src/docs/product/performance/trace-view.mdx

Currently, the trace view will only show up to 100 transactions from the trace. Additional transactions can be found through [Discover](/product/discover-queries).

## Trace View

The trace view provides a split view. On the left, you can view the trace's transaction tree. On the right, you can view a representation of each transaction as a colored rectangle. The first row lists the trace itself, with all other spans descending from that trace.

You can display these views from:
- [Transaction Summary](/product/performance/transaction-summary/): Select the [Performance Homepage](/product/performance/), the affected transaction, then click a trace ID.
- [Event Details](/product/performance/event-detail/): Click the trace ID beneath the quick trace on the top right.
- [Issue Details](/product/issues/issue-details/): Click "View Full Trace" above the quick trace.
- [Query Builder](/product/discover-queries/query-builder/): Build a query with the trace column, then click a trace ID in the table.

Trace IDs are linked to the corresponding Trace Detail.

## Waterfall

> The waterfall view is a split view. On the left, you can view the trace's transaction tree. On the right, each transaction displays as a horizontal bar.

Reviewer: Reading this again and this section sounds like a repeat of the above Trace View section.

Author: Hmm good point, let me remove this

## Transaction Details

Clicking a transaction row expands the details of that transaction to view many of the attached properties, such as tags and other field data. With the transaction name, you can view the [Transaction Summary](/product/performance/transaction-summary/) of that transaction across traces. Click "View Summary" to see that summary. As well with the transaction ID, you can see the [event details](/product/performance/event-detail/) of that transaction by clicking on "View Transaction".

# Troubleshooting

## Orphan Traces and Broken Subtraces

> In a fully instrumented system, a span in one service will connect to a transaction in a subsequent service. For a variety of reasons, a transaction in the first service may not arrive in Sentry. When Sentry encounters these types of transactions within a trace, we link the transactions with a dashed line since they can no longer be directly connected to the root.

Reviewer: Transactions from any service have a chance of not arriving in Sentry, not just the first.

```suggestion
In a fully instrumented system, a span in one service will connect to a transaction in a subsequent service. For a variety of reasons, a transaction in a service may not arrive in Sentry. When Sentry encounters these types of transactions within a trace, we link the transactions with a dashed line since they can no longer be directly connected to the root.
```

Broken subtraces may be caused by:
- [Ad blockers](/platforms/javascript/troubleshooting/#dealing-with-ad-blockers) may prevent transactions in browsers being sent, but HTTP requests to backend projects will still create child transactions
- [Rate-limiting](/product/accounts/quotas/#limiting-events) on a project may cause only some events to be sent to Sentry
- [Project permissions](/product/accounts/membership/#restricting-access) may mean you do not have access to transactions in another project

## Multiple Roots

> Each trace ID should have only one transaction without any parents -- the root. Included instrumentation should prevent multiple roots; however, if the trace ID of your transactions are being set using custom instrumentation, you may encounter multiple roots.

Reviewer: Something like this? Because there can be multiple transactions without any parents (multiple orphans case), but multiple roots happen when there are more than 1 top-level parent.

```suggestion
Each trace ID should have only one transaction with a top-level parent -- the root. Included instrumentation should prevent multiple roots; however, if the trace ID of your transactions is being set using custom instrumentation, you may encounter multiple roots.
```

Author: Hmm not sure if I understand what a `top-level parent` is in the context of transactions.

Reviewer: Sorry, a top-level parent is just referring to the root (probably better since the docs explain what a root is).

Now that I re-read this, I think my confusion around the sentence is specifically around the grammar. Can we change it from

> Each trace ID should have only one transaction without any parents -- the root.

to something like

> Each trace ID should have only one root, a trace without any parents.

---
title: "Trace View"
sidebar_order: 7

> description: "Learn more about the trace view and what kind of high level information you can see for a Trace"

Reviewer 2: ```suggestion
description: "Learn more about the trace view and the high level information you can view for a Trace."
```

---

<Note>

This feature is available only if you're in the Early Adopter program. Features available to Early Adopters are still in-progress and may have bugs. We recognize the irony.

If you’re interested in being an Early Adopter, you can turn your organization’s Early Adopter status on/off in General Settings. This will affect all users in your organization and can be turned back off just as easily.

</Note>

From Performance, Issues, and Discover, you can drill down into a single [trace](/product/performance/distributed-tracing) and traverse all of the transactions in that trace. This will accelerate your ability to debug slow services, identify related errors, and root out other bottlenecks.

Currently, the trace view will only show up to 100 transactions from the trace. Additional transactions can be found through [Discover](/product/discover-queries).

> 

Reviewer 2: delete extra line

## Trace View

> The trace view is a split view where the left-hand side shows the trace's transaction tree, and the right-hand side represents each transaction as a colored rectangle. The first row listed is always the trace itself, from which all other spans in the trace descent.

Reviewer 2: ```suggestion
The trace view provides a split view. On the left, you can view the trace's transaction tree. On the right, you can view a representation of each transaction as a colored rectangle. The first row lists the trace itself, with all other spans descending from that trace.
```

> to find these views, you can either go through the [Transaction Summary](/product/performance/transaction-summary/), [Event Details](/product/performance/event-detail/), [Issue Details](/product/issues/issue-details/), or [Query Builder](/product/discover-queries/query-builder/). Trace IDs will be linked to the corresponding Trace Detail.
> 
> ## With Transaction Summary
> 
> Select the [Performance Homepage](/product/performance/), then the affected transaction, and clicking on a trace ID in the table will display the trace data.
> 
> ## With Event Details
> 
> Beneath the quick trace on the top right of the Event Details page, click on the trace ID will display the trace data.
> 
> ## With Issue Details
> 
> Above the quick trace on the Issue Details page, click on View Full Trace will display the trace data.
> 
> ## With Query Builder
> 
> Build a query with the trace column, then clicking a trace ID in the table will display the trace data.

Reviewer 2: ```suggestion
You can display these views from:
- [Transaction Summary](/product/performance/transaction-summary/): Select the [Performance Homepage](/product/performance/), the affected transaction, then click a trace ID.
- [Event Details](/product/performance/event-detail/): Click the trace ID beneath the quick trace on the top right.
- [Issue Details](/product/issues/issue-details/): Click "View Full Trace" above the quick trace.
- [Query Builder](/product/discover-queries/query-builder/): Build a query with the trace column, then click a trace ID in the table.

Trace IDs are linked to the corresponding Trace Detail.
```

## Waterfall

> The waterfall view is a split view where the left reflects the trace's transaction tree, and the right reflects each transaction as a horizontal bar.

Reviewer 2: ```suggestion
The waterfall view is a split view. On the left, you can view the trace's transaction tree. On the right, each transaction displays as a horizontal bar.
```

## Transaction Details

> Clicking on a transaction row expands the details of that span. From here you can see many of the attached properties, such as tags and other field data. With the transaction name, you'll be able to view the [Transaction Summary](/product/performance/transaction-summary/) of that transaction across traces. Click on "View Summary" to see that summary.

Reviewer 2: ```suggestion
Clicking a transaction row expands the details of that span to view many of the attached properties, such as tags and other field data. With the transaction name, you can view the [Transaction Summary](/product/performance/transaction-summary/) of that transaction across traces. Click "View Summary" to see that summary.
```

## Traversing Transactions

> Every transaction within a trace has additional details such as spans that. Click on "View Transaction" to see a detailed view of that transaction.

Reviewer 2: I think this sentence is incomplete.

# Troubleshooting

## Orphan Traces and Broken Subtraces

> While in a fully instrumented system, a span in one service will give rise to a transaction in a subsequent service. There can be a variety of reasons why the transaction in the first service may not arrive in Sentry. When Sentry encounters transactions like this within a trace we link the transactions with a dashed line since they can no longer be directly connected to the root.

Reviewer 2: ```suggestion
In a fully instrumented system, a span in one service will connect to a transaction in a subsequent service. For a variety of reasons, a transaction in the first service may not arrive in Sentry. When Sentry encounters these types of transactions within a trace, we link the transactions with a dashed line since they can no longer be directly connected to the root.
```

> Some reasons for trace data like this include:

Reviewer 2: ```suggestion
Transactions without a root may be caused by:
```

Author: Switched this to `Broken subtraces may be caused by:`

- [Ad blockers](/platforms/javascript/troubleshooting/#dealing-with-ad-blockers) may prevent transactions in browsers being sent, but HTTP requests to backend projects will still create child transactions

> - [Rate-limiting](/product/accounts/quotas/#limiting-events) on a project can mean only some of the events arrive in Sentry

Reviewer 2: ```suggestion
- [Rate-limiting](/product/accounts/quotas/#limiting-events) on a project may cause only some events to be sent to Sentry
```

> - [Project permissions](/product/accounts/membership/#restricting-access) could result in transactions in another project to appear missing if your account does not have access

Reviewer 2: ```suggestion
- [Project permissions](/product/accounts/membership/#restricting-access) may mean you do not have access to transactions in another project 
```

## Multiple Roots

> For each trace id there should exist only one transaction without any parents, included instrumentation should prevent this case but if the trace id of your transactions are being set manually you may see this problem.

Reviewer 2: ```suggestion

Each trace ID should have only one transaction without any parents -- the root. Included instrumentation should prevent multiple roots; however, if the trace ID of your transactions are being set using custom instrumentation, you may encounter multiple roots.
```
